### ISSUE TITLE
Navigation button intention could be clearer

### ISSUE THREAD

**Issue by tim-hm** [canonical/ubuntu-desktop-installer#1585]
## Issue

The navigation buttons intent could be clearer.

## Discussion

1. The "Continue" button styling is commonly used to indicate a disabled button, not a 'next' step (ref [material ui's disabled button state](https://mui.com/material-ui/react-button/#contained-button), the [yaru theme disabled button state](https://github.com/ubuntu/yaru.dart), or just a standard disabled HTML button). 

2. If we want language symmetry then the opposite of "Back" is not "Continue". "Forward" might be more appropriate. 

3. When the back button is selected using keyboard navigation it the two buttons look (almost) the same meaning its unclear what happens when pressing Enter:

<img src="https://user-images.githubusercontent.com/1335525/224334755-d47b0e9b-f44c-4b1b-8611-ad49758f7fa3.png" width=300 />

4. I appreciate that we should differentiate between the "Install" button resulting in system changes and the "change page" buttons.

5. The final install button is green with the default orange accent colour, but if the accent colour changes in system settings then the "Install" button follows the new system accent colour. The left button is the default button without accent changes and the right button is the button after applying the green style:

<img src="https://user-images.githubusercontent.com/1335525/224340409-e9f8d25f-0a1e-4407-9126-6deebd719edd.png" width=300 />

6. Inside modals the 'primary' or selected button is subtly different, especially when using keyboard navigation (see point 3 above):

<img src="https://user-images.githubusercontent.com/1335525/224341180-8ad302fc-075c-40a4-80a4-a8737e90bb9f.png" width=300 />

## Recommendations

1. Use a contained button using a white(ish) background and accent colour on the text (if the yaru package permits). Alternatively, use the accent-colour-filled contained button for the Next/Continue step. For example:

<img src="https://user-images.githubusercontent.com/1335525/224339707-3b631b29-a8ae-482c-9525-c3afd31ed94c.png" width=300 />

2. For modals use the same styling as the "Next" or "Continue" button to improve keyboard selection indication and highlight the 'primary' button.

3. Use a hard-coded color button for the final "Install" to communicate its non-reversible action and distinguish it from all other buttons in the installer. Red might be a good choice for this.

4. Consider "Back" and "Forward" or "Previous" and "Next" for the button text.

**Comment by Feichtmeier** [comment]
@tim-hm I also think that the grey filled buttons do not fit perfectly to the importance of the action of "go next"

How about using a half transparent green instead? With the last page having a fully green button?
<img width="596" alt="grafik" src="https://user-images.githubusercontent.com/15329494/224936933-af2e4649-cb34-4e0a-bb39-25b0c3cde54d.png">

The reason why yaru uses a green button as the fully colored button type is that the ubuntu orange looks kind of aggressive when used on a big area/surface, even if it looks fine when used as accent or little "gem" elsewhere. 
But yaru.dart should not limit the design in the installer IMO and flutter allows for easy design adjustments wherever needed

**Comment by tim-hm** [comment]
I think that makes sense as I also didn't like the look of the filled orange variant. 

Should the "go next" button follow the system accent colour or be independent?

**Comment by Feichtmeier** [comment]
The default theme in yaru is the only one of all variants that has two accent colors (orange and green)
So we could stick to this by using the green for the default theme with the lowered opacity
<img width="912" alt="Bildschirm­foto 2023-03-14 um 09 42 17" src="https://user-images.githubusercontent.com/15329494/224944600-6fd20089-42c6-45ae-a2cc-9c39484f79af.png">

And for all other variants the primary color could be used
<img width="912" alt="Bildschirm­foto 2023-03-14 um 09 43 04" src="https://user-images.githubusercontent.com/15329494/224944757-145b53ac-e3f9-4d5b-8bae-2796bb564ce8.png">
<img width="912" alt="Bildschirm­foto 2023-03-14 um 09 43 46" src="https://user-images.githubusercontent.com/15329494/224944906-9f0d66c4-d452-48d0-a8e9-92c0a3379b5b.png">

**Comment by elioqoshi** [comment]
> The "Continue" button styling is commonly used to indicate a disabled button, not a 'next' step.

The color contrast here is quite different from a disabled button, however I see what you mean that it might look slightly disabled. One way to improve that is too use a lighter fill color here.

I don't think using primary color green or something which brings more attention here is appropriate for a neutral step going forward.

@jpnurmi would we be able to use a slightly lighter gray?

**Comment by elioqoshi** [comment]
> If we want language symmetry then the opposite of "Back" is not "Continue". "Forward" might be more appropriate.

Yeah I am not sure it's ideal right now. But I wouldn't prioritize language symmetry over usual user perceptions and UX copy practices. I don't think "Forward" is appropriate as I could not benchmark that with any other application wizard or flow using that. 

I had a quick look at some sources:
https://www.nngroup.com/articles/wizards/
https://uxplanet.org/wizard-design-pattern-8c86e14f2a38

This is how Patternfly (Red Hat does it):
https://www.patternfly.org/v4/components/wizard/design-guidelines/

![grafik](https://user-images.githubusercontent.com/5436686/225236109-3e543499-f140-450b-acad-3825bd3669aa.png)

Not sure why their places are interchanged however.

Considering all that, I'd propose:

1. Back & Next
2. Previous & Next

I will check with my UX colleagues to see how we deal with this in other Canonical products to keep it consistent. What do you think @tim-hm ?

**Comment by elioqoshi** [comment]
> When the back button is selected using keyboard navigation it the two buttons look (almost) the same meaning its unclear what happens when pressing Enter:

I'd consider this a bug, meaning the focus state of the button is not suitable. @jpnurmi What focus state effect do we currently have? I don't see a focus state in the Yaru button examples. Usually focus states surround the button instead of changing its look.

**Comment by elioqoshi** [comment]
> The final install button is green with the default orange accent colour, but if the accent colour changes in system settings then the "Install" button follows the new system accent colour. The left button is the default button without accent changes and the right button is the button after applying the green style:

This is a pretty big philosophical issue but I agree with you. So the issue is that in the default theme we use green because it's a semantic signal color. In the theme accent variants, the specific accent colors are used instead, losing its semantic meaning. 

While this is not a simple issue, as this is an upstream Yaru discussion my current gut feeling is siding with:

- Primary Success Buttons (like this one here) should always be green and not be dictated by the accent choice.
- There could be alternative accent color-dictated buttons for when semantics (positive or negative) are not relevant.

Regarding this we need to talk with the Yaru folks and get their input here in order to determine if this change will happen in Yaru, or whether we just do it on the installer and not upstream for the time being and revisit this topic later.

**Comment by elioqoshi** [comment]
Moving this to a future milestone so we can track it for later as there are some really good points mentioned here.

**Comment by elioqoshi** [comment]
After some intense discussions here and some further research, I advise we should go for Back and Next.

I asked Sophie, our Content Writer on some additional advice on that. 

> So for the installer nav, which is a stepped process the user is following in order to complete a particular important action, we want to support them to go back if they're feeling nervous or want to correct something. we want to urge them ahead, without pressure, to the next step of the process. but of course, they can always go back until they install.

I think seeing it in this light is very sensible and coupled with the fact that macOS and Windows use Back too as well as Windows using Next (macOS uses Continue however).

**Comment by elioqoshi** [comment]
Windows:

![image](https://github.com/canonical/ubuntu-desktop-installer/assets/5436686/911614b2-c8ac-4d69-92b2-a8babe565acc)


![image](https://github.com/canonical/ubuntu-desktop-installer/assets/5436686/f4fbd20e-3431-47c8-82e0-359c09545ff0)

![image](https://github.com/canonical/ubuntu-desktop-installer/assets/5436686/781934ec-1407-419c-a23a-64b2a9549e37)

![image](https://github.com/canonical/ubuntu-desktop-installer/assets/5436686/c8e280d7-333b-44a8-9c67-9349165938d6)

**Comment by mwhudson** [comment]
FWIW the server tui is not very consistent about this, would be nice to clean that up too.

**Comment by elioqoshi** [comment]
@mwhudson Yeah, so many things, so few hands!
